### PR TITLE
Use optional args for JSON.stringify and JSON.parseExn

### PR DESCRIPTION
**Pull request by cknitt**
This would simplify the API a lot, but it would be a breaking change.

What do you think?

### CODE REVIEW

**Comment by zth** [comment]
Love it! If we keep the old functions around for a while but deprecate them then we can avoid breaking, right?

**Comment by cknitt** [comment]
> Love it! If we keep the old functions around for a while but deprecate them then we can avoid breaking, right?

Yes, will do that! (And also update the documentation and add some tests.)

**Comment by cknitt** [comment]
@zth BTW, [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#space) and TypeScript typedefs call the last parameter `space` instead of `indent`. I think we should do that, too.

**Comment by cknitt** [comment]
@zth Ready for re-review, could you have a look?

File: src/Core__JSON.resi [hunk 1]
```diff
@@ -31,6 +35,28 @@ try {
 } catch {
 | Exn.Error(_) => Console.log("error")
 }
+
+let reviver = (_, value) => {
+  let valueType = JSON.Classify.classify(value)
```
Comment (zth) [inline src/Core__JSON.resi:40]: Does this really need classify? Can't it use the Json representation directly?

**Review by zth — APPROVED** [review]
This looks good to me. 2 comments:
- Docstrings should probably be updated to match the new v11 Json representation.
- Will this require 11.1 to play nicely with undefineds?

**Review by zth — APPROVED** [review]

**Pull request merged**